Mathlib/Tactic/FProp/ToStd.lean

```diff
@@ -0,0 +1,195 @@
+/-
+Copyright (c) 2024 Tomas Skrivan. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Tomas Skrivan
+-/
+import Lean
+
+import Std.Data.Nat.Lemmas
+import Std.Lean.Expr
+
+
+/-!
+## `fprop` missing function from standard library
+-/
```

Comment:
The definitions here could:
* go out into Mathlib, into the existing files about `Expr`, `Array`, etc (as part of this PR)
* go up to Std (but there is additional overhead from having multiple PRs!)
* stay put in this file

I think in practice all options are viable for this PR.

Reply:
Ok, I will go over the file, add documentation and see what would be fitting to move to corresponding mathlib or std files. Some of the stuff might already be somewhere, it was just faster to write it myself.